tokio/src/runtime/task/core.rs

```diff
@@ -182,6 +184,10 @@ pub(crate) struct Header {
     /// The tracing ID for this instrumented task.
     #[cfg(all(tokio_unstable, feature = "tracing"))]
     pub(super) tracing_id: Option<tracing::Id>,
+
+    /// Custom user defined metadata for this task for use in hooks.
+    #[cfg(tokio_unstable)]
+    pub(super) user_data: UserData,
```

Comment:
I believe the task header is supposed to fit in 64 bytes. Is that still the case? I'm thinking we might need to put this in the trailer.

Reply:
Yep, it's still the case:

```
$ cargo clean; RUSTFLAGS="--cfg tokio_unstable" cargo +nightly rustc --lib --all-features -- -Zprint-type-sizes | grep '`runtime::task::core::Header`'
...
print-type-size type: `runtime::task::core::Header`: 56 bytes, alignment: 8 bytes
```

and reviewing the full output confirming that's with the additional user_data, with its expected 16 bytes:
```
print-type-size type: `runtime::task::core::Header`: 56 bytes, alignment: 8 bytes
print-type-size     field `.state`: 8 bytes
print-type-size     field `.queue_next`: 8 bytes
print-type-size     field `.vtable`: 8 bytes
print-type-size     field `.owner_id`: 8 bytes
print-type-size     field `.tracing_id`: 8 bytes
print-type-size     field `.user_data`: 16 bytes
```